app/src/org/commcare/android/database/connect/models/ConnectJobRecord.java

```diff
@@ -416,14 +417,12 @@ public Hashtable<String, Integer> getDeliveryCountsPerPaymentUnit(boolean todayO
         Hashtable<String, Integer> paymentCounts = new Hashtable<>();
         for(int i = 0; i < deliveries.size(); i++) {
             ConnectJobDeliveryRecord delivery = deliveries.get(i);
-            if(!todayOnly || sameDay(new Date(), delivery.getDate())) {
-                int oldCount = 0;
-                if (paymentCounts.containsKey(delivery.getSlug())) {
+            int oldCount = 0;
+            boolean isSameDay=sameDay(new Date(),delivery.getDate());
+                if(paymentCounts.containsKey(delivery.getSlug()) && (isSameDay && todayOnly)) {
```

Comment:
Pretty sure the logic here is still incorrect. For example, if todayOnly=false (i.e. we want the all-time counts), the code in the if-statement never gets called.

I wasn't sure whether we'd get this ticket finished before the deadline for QA and wanted to make sure at least this bug in the code was fixed, so in the meantime I added what I believe is a correct fix which you can see [here](https://github.com/dimagi/commcare-android/blob/97c31019f963f2821e69359d16a225449fc193c7/app/src/org/commcare/android/database/connect/models/ConnectJobRecord.java#L437C13-L444C14).

The idea is to only consider records when either:
- We want all-time (i.e. todayOnly=false)
OR
- The record is from today

If either check above passes, then the record is sorted in to paymentCounts.